CONVENTIONS.md

```diff
@@ -116,12 +116,85 @@ Name screen coordinates `x` and `y`, rather than `left` and `top`.
+Write sentences for class, method and fields javadocs, starting with an uppercase and ending with a period. Start method docs with verbs, like `Gets` or `Called`. Use HTML tags such as `<p>` if the docs have several paragraphs, as line wraps are converted to spaces in the generated documentation. Feel free to start a new line whenever you feel the current line is too long.
 
-Write quick descriptions for parameter javadocs as well as `@return` tags, with no uppercase or period. Add parameter docs to the parameter itself instead of using the `@param` tag.
+Parameter and `@return` documentation should use quick descriptions without initial capitalization or punctuation, such as `whether the block placement was successful`.
+
+Avoid using the `@param` tag for documentation of owner methods. Add parameter docs to the parameter itself instead of adding `@param` tags to its owner method. However, you can use the tag for type parameters (such as `<T>` in `public class Lazy<T>`), who don't have mapping entries for their documentations.
+
+Javadoc will take the first sentence, ended by the first `.`, as a brief description of the member you are documenting. If you need to enclose another sentence to serve as a description, such as when you have multiple periods within a sentence, enclose the summary within the `{@summary }` inline tag. An example sentence with multiple periods enclosed within a summary tag would be `{@summary Returns the client's player, i.e. the controlled player.}`.
+
+### Tooling
+
+Fabric-hosted Javadocs are generated using [JDK 11 Standard Doclet](https://docs.oracle.com/en/java/javase/11/docs/specs/doc-comment-spec.html) and can use any feature it supports, though you can personally build the documentation with a newer Java version. See [the 'Checking Javadoc' section](#checking-javadoc) for how to build the documentation locally.
+
+### Custom tags
+
+A few additional block tags are supported:
+
+ - `@apiNote`: API Notes. A few comments for users of the documented member.
+ - `@implSpec`: Implementation Specification. Tells how this method should be implemented.
+ - `@implNote`: Implementation Notes. A few comments on the implementation.
+
+Feel free to use these tags and write under these sections.
+
+### Linking to other members
 
 Use `@link`, `@linkplain` and `@see` tags to refer to other parts of the code.
 
+You can use the simple name of a class in the following scenarios:
+
+- If it is from the `java.lang` package
+- If it is from the same package as the currently documented class
+- If it is part of the signature of the class or its members (methods and fields)
+
+If it does not fulfill one of these scenarios, use the [full binary name](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/ClassLoader.html#binary-name), such as `com.google.common.collect.Lists` rather than simply `Lists`. Unlike class naming in enigma, do not use `/` to separate packages; use `.` instead.
```

Comment:
```suggestion
If it does not fulfill one of these scenarios, use the [full binary name](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/ClassLoader.html#binary-name), such as `com.google.common.collect.Lists` rather than simply `Lists`. The separator is `.`, unlike class naming in Enigma, which uses `/`.
```

Reply:
Nah, my emphasis is on not using `/`.